lib/Sema/ResilienceDiagnostics.cpp

```diff
@@ -231,7 +231,7 @@ static bool diagnoseValueDeclRefExportability(SourceLoc loc, const ValueDecl *D,
                                               const ExportContext &where) {
   assert(where.mustOnlyReferenceExportedDecls());
 
-  auto definingModule = D->getModuleContext();
+  auto definingModule = D->getModuleContextForNameLookup();
```

Comment:
Just to clarify: this change doesn't actually affect the semantic checking itself, right? It just helps to give clearer diagnostics that point to the actual Clang module rather than the `__ObjC` module.